lua/bufferline/offset.lua

```diff
@@ -15,6 +15,9 @@ local t = {
 ---@return string
 local function get_section_text(size, highlight, offset)
   local text = offset.text
+  if offset.text_type == "function" then
+    text = _G[text]()
+  end
```

Comment:
@Akianonymus this doesn't need to be specified as a separate field, and use lua's `type` argument to check the type of the field. I suggest
```suggestion
if type(text) == "function" then
 text = text()
end
```

Reply:
There's also no need to reference the global object in lua since that will point at a non-existent text field and cause errors.